pages/index.tsx

```diff
@@ -1,70 +1,70 @@
+import '@reach/dialog/styles.css'
 import type { NextPage } from 'next'
-import Head from 'next/head'
-import Image from 'next/image'
 import styles from '../styles/Home.module.css'
+import {Dialog} from '@reach/dialog'
+import { useState } from 'react'
 
-const Home: NextPage = () => {
-  return (
-    <div className={styles.container}>
-      <Head>
-        <title>Create Next App</title>
-        <meta name="description" content="Generated by create next app" />
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
-
-      <main className={styles.main}>
-        <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.js!</a>
-        </h1>
+function LoginForm({onSubmit, buttonText}){
+  function handleSubmit(event: React.FormEvent<HTMLInputElement>){
+    event.preventDefault()
+    const {username, password} = event.target.elements
 
-        <p className={styles.description}>
-          Get started by editing{' '}
-          <code className={styles.code}>pages/index.tsx</code>
-        </p>
+    onSubmit({
+      username: username.value,
+      password: password.value
+    })
+  }
+  return(
+    <form onSubmit={handleSubmit}>
+      <div>
+        <label htmlFor="username">Username</label>
+        <input id="username" type="text" />
+      </div>
+      <div>
+        <label htmlFor="password">Password</label>
+        <input id="password" type="text" />
+      </div>
+      <div>
+        <button type="submit">{buttonText}</button>
+      </div>
+    </form>
+  )
+}
 
-        <div className={styles.grid}>
-          <a href="https://nextjs.org/docs" className={styles.card}>
-            <h2>Documentation &rarr;</h2>
-            <p>Find in-depth information about Next.js features and API.</p>
-          </a>
+const Home: NextPage = () => {
+  const [openModal, setOpenModal] = useState('none')
```

Comment:
Can you supply a type instead? This will allow others including ourselves to understand what should be in here.

We should use `boolean` when it only requires 2 possibilities.

```suggestion
  const [openModal, setOpenModal] = useState<boolean>(false)
```